sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -484,6 +484,15 @@ class StreamExecution(
      """.stripMargin
   }
 
+  private def toInfo: ContinuousQueryInfo = {
```

Comment:
Do we need to make this a deep copy, especially for `sourceStatuses` and `sinkStatus` ?